src/mqt/problemsolver/csp.py

```diff
@@ -1,16 +1,32 @@
 from __future__ import annotations
 
+from typing import TYPE_CHECKING, TypedDict
+
 import numpy as np
-from mqt import ddsim
+from mqt.ddsim import DDSIMProvider
 from qiskit import QuantumCircuit, QuantumRegister, execute
 
+if TYPE_CHECKING:
+    from qiskit.circuit import Instruction
+
+
+class Constraint(TypedDict, total=False):
+    """
+    Class to store the properties of a single constraint.
+    """
+
+    constraint_type: str
+    operand_one: str
+    operand_two: str
+    to_be_satisfied_sum: int
+
 
 class CSP:
     def solve(
         self,
-        constraints: list[dict],
-        quantum_algorithm="Grover",
-    ):
+        constraints: list[Constraint],
+        quantum_algorithm: str = "Grover",
+    ) -> tuple[int, int, int, int] | bool:
```

Comment:
Again, nothing for this PR, but to keep in mind for the future: If we really want to expand upon this, dedicated classes or TypedDicts could be employed to make all of this a little more general. 
That holds for the return type here but also for all the tuples of registers down below.
Feel free to just resolve this conversation, but keep the idea in mind.